tests/api/route-services/work-response.test.js

```diff
@@ -173,93 +141,71 @@ describe('tests for the work-response service', () => {
           type: 'inline',
         }],
     };
+
     const w = new WorkResponseService(io, workResponse);
     const expectedResponse = JSON.parse(JSON.stringify(workResponse));
     expectedResponse.results
       .forEach((res) => delete res.type);
 
-    w.handleResponse().then(() => {
-      expect(toSpy).toHaveBeenCalledTimes(1);
-      expect(toSpy).toHaveBeenCalledWith('mysocketid');
-      expect(emitSpy).toHaveBeenCalledTimes(1);
-      expect(emitSpy).toHaveBeenCalledWith('WorkResponse-55', expectedResponse);
-      return done();
+    client.on(`WorkResponse-${workResponse.request.uuid}`, (res) => {
+      expect(res).toEqual(expectedResponse);
+      done();
     });
+
+    await w.handleResponse();
   });
 
   it('Can consume work response with a single processS3PathType item', async (done) => {
+    expect.assertions(3);
+
     const workResponse = {
-      request: {
-        uuid: '55',
-        socketId: 'mysocketid',
-        experimentId: '5e959f9c9f4b120771249001',
-        timeout: '2099-01-01T00:00:00Z',
-        body: {
-          name: 'GetEmbedding',
-          type: 'pca',
-        },
-      },
-      response: {
-        cacheable: true,
-        error: false,
-      },
+      ...responseFormat,
       results: [
         {
           'content-type': 'application/json',
           'content-encoding': 'utf-8',
-          body: 'worker-results-staging/509520fe-d329-437d-8752-b5868ad59425/48637d30-a88d-481e-b8cc-5eedd7e3af1c',
+          body: `${S3_BUCKET_NAME}/${S3_RESULT_ID}`,
           type: 's3-path',
         },
       ],
     };
+
     AWSMock.mock('S3', 'getObject', (params, callback) => {
-      getObjectSpy(params);
-      callback(null, s3Object);
+      const { Bucket, Key } = params;
+
+      expect(Bucket).toEqual(S3_BUCKET_NAME);
+      expect(Key).toEqual(S3_RESULT_ID);
+
+      callback(null, {
+        ContentEncoding: 'utf-8',
+        ContentType: 'application/json',
+        Body: OBJECT_CONTENT,
+      });
     });
-    const w = new WorkResponseService(io, workResponse);
-    const expectedResponse = JSON.parse(JSON.stringify(workResponse));
-    delete expectedResponse.results[0].type;
-    expectedResponse.results[0].body = 'my amazing body';
-
-    w.handleResponse().then(() => {
-      expect(toSpy).toHaveBeenCalledTimes(1);
-      expect(toSpy).toHaveBeenCalledWith('mysocketid');
-      expect(emitSpy).toHaveBeenCalledTimes(1);
-      expect(emitSpy).toHaveBeenCalledWith('WorkResponse-55', expectedResponse);
-      expect(getObjectSpy).toHaveBeenCalledTimes(1);
-      expect(getObjectSpy).toHaveBeenCalledWith(
-        {
-          Bucket: 'worker-results-staging',
-          Key: '509520fe-d329-437d-8752-b5868ad59425/48637d30-a88d-481e-b8cc-5eedd7e3af1c',
-          ResponseContentType: 'application/json',
-          ResponseContentEncoding: 'utf-8',
-        },
-      );
```

Comment:
Again, instead of these old spies, we can just assert directly inside the actual mock, no need to forward to a `getObjectSpy`.